platform/packages/platform/src/response.rs

```diff
@@ -31,7 +31,7 @@ where
 
     events
         .into_iter()
-        .fold(cw_resp, |res, event| res.add_event(event.into()))
+        .fold(cw_resp, |res, event| res.add_event(event))
```

Comment:
This should do the trick as `self` is taken by value.
```suggestion
        .fold(cw_resp, CwResponse::add_event)
```

I'd suggest considering this for the code constructing `cw_resp` too.